### PR TITLE
Fixes #513

### LINKED ISSUE #513: Two Factor - Email - User model doesnt get init properly
### What steps will reproduce the problem?
Enable Two-Factor Email. Logout. Try Loging in - Error Message

### What is the expected result?
Email to be sent with the code

### What do you get instead?
Error message - Unable to send email

### PR DESCRIPTION
**Pull request by deadmantfa**
| Q             | A
| ------------- | ---
| Is bugfix?    | yes
| New feature?  | no
| Breaks BC?    | no
| Tests pass?   | no
| Fixed issues  | comma-separated list of tickets # fixed by the PR, if any

### CODE REVIEW

File: src/User/Service/MailService.php [hunk 1]
```diff
@@ -83,11 +83,20 @@ public function getType()
...
+                ->setFrom($this->from)
+                ->setTo($this->to)
+                ->setSubject($this->subject)
+                ->send();
+
+            if (!$result) {
+                Yii::error("Email sending failed to '{$this->to}'.", 'mailer');
+            }
+            return $result;
+        } catch (\Exception $e) {
```
Comment (maxxer) [inline src/User/Service/MailService.php:98]: Why this catch? Looks like you're trying to capture a configuration error here

File: src/User/Validator/TwoFactorEmailValidator.php [hunk 2]
```diff
@@ -111,6 +111,6 @@ public function getUnsuccessLoginMessage($codeDurationTime)
     */
     public function generateCode()
     {
-        return $this->make(TwoFactorEmailCodeGeneratorService::class, $this->user)->run();
+        return $this->make(TwoFactorEmailCodeGeneratorService::class, [$this->user])->run();
```
Comment (maxxer) [inline src/User/Validator/TwoFactorEmailValidator.php:114]: 👍

**Review by maxxer — CHANGES_REQUESTED** [review]
Hi, thank you for your PR! The call to `TwoFactorEmailCodeGeneratorService` seems correct, but I'm unsure about the `try / catch` in MailService, I'd rather avoid merging that change

File: src/User/Service/MailService.php (hunk 1 above)
Comment (deadmantfa) [reply src/User/Service/MailService.php:98]: The issue here was that the error was not clear as the email wasn't being sent correctly. With this we can find what the actual issue is.

In my case the $this->user was not set and in turn $this->to was set to blank

Comment (deadmantfa) [reply src/User/Service/MailService.php:98]: If required i will remove the try catch.

**Review by TonisOrmisson — COMMENTED** [review]
instead of changging mailService, you could use the MailAwareTrait like its used here

https://github.com/2amigos/yii2-usuario/blob/737d6c1894ed584784b195e4a856fb734009eeb6/src/User/Service/ResendConfirmationService.php#L38

**Comment by deadmantfa** [comment]
Oh will have a look. I did not see this. I encountered an issue so just fixed it. Will make the change i encounter another issue for which i will create another PR

**Comment by deadmantfa** [comment]
If these changes are good. Can we publish them. I need to use it in my live environment ;P

**Comment by maxxer** [comment]
Please put the changelog entry in the existing `dev` section :)

**Pull request closed without merging**